velox/functions/lib/Re2Functions.cpp

```diff
@@ -391,6 +391,44 @@ bool matchExactPattern(
       std::memcmp(input.data(), pattern.data(), length) == 0;
 }
 
+// Match the input(from the position of start) with relaxed pattern(from the
+// specified subPattern).
+bool matchRelaxedFixedPattern(
+    StringView input,
+    const PatternMetadata& patternMetadata,
+    size_t start) {
+  // Compare the length first.
+  if (input.size() - start < patternMetadata.length()) {
+    return false;
+  }
+
+  // Compare each literal range.
+  for (auto i = 0; i < patternMetadata.numLiteralRanges(); i++) {
```

Comment:
nit: Wondering if the code can be simplified by iterating over the vector of ranges.

```
for (const auto& range : patternMetadata.literalRanges()) {
   const auto start = range.first;
   const auto length = range.second;
   ....memcmp...
}
```